Lib/test/test_msvcrt.py

```diff
@@ -61,34 +61,48 @@ def test_get_osfhandle(self):
 
 
 class TestConsoleIO(unittest.TestCase):
+    def run_in_separated_process(self, code):
+        # Run test in a seprated process to avoid stdin conflicts.
+        # See: gh-110147
+        cmd = [sys.executable, '-c', code]
+        subprocess.run(cmd, check=True, capture_output=True,
+                       creationflags=subprocess.CREATE_NEW_CONSOLE)
+
+
     def test_kbhit(self):
-        h = msvcrt.get_osfhandle(sys.stdin.fileno())
-        flush_console_input_buffer(h)
-        self.assertEqual(msvcrt.kbhit(), 0)
+        code = dedent('''
+            import msvcrt
+            assert msvcrt.kbhit() == 0
+        ''')
+        self.run_in_separated_process(code)
 
     def test_getch(self):
         msvcrt.ungetch(b'c')
         self.assertEqual(msvcrt.getch(), b'c')
 
     def test_getwch(self):
-        with open('CONIN$', 'rb', buffering=0) as stdin:
-            h = msvcrt.get_osfhandle(stdin.fileno())
-            flush_console_input_buffer(h)
-
-            write_input(stdin, c_encoded)
-            self.assertEqual(msvcrt.getwch(), c)
+        code = dedent(f'''
+            import msvcrt
+            from _testconsole import write_input
+            with open("CONIN$", "rb", buffering=0) as stdin:
+                write_input(stdin, {repr(c_encoded)})
+                assert msvcrt.getwch() == "{c}"
+        ''')
+        self.run_in_separated_process(code)
 
     def test_getche(self):
         msvcrt.ungetch(b'c')
         self.assertEqual(msvcrt.getche(), b'c')
```

Comment:
It may be safer to run this test in a subprocess with a new console, no?

Reply:
Would CREATE_NO_WINDOW flag prevent popups?

> The process is a console application that  is being run without a console window. Therefore, the console handle for  the application is not set. This flag is ignored if the application is not a console application, or if it is used with either CREATE_NEW_CONSOLE or DETACHED_PROCESS.

says: https://learn.microsoft.com/en-us/windows/win32/procthread/process-creation-flags

Reply:
Seems it should work, I'll try it.

Reply:
> the console handle for the application is not set

Means no console and the test is irrelevant.

> ignored if ... used with either CREATE_NEW_CONSOLE

This is the flag that you'd use to make the test more reliable, and it's incompatible with `CREATE_NO_WINDOW`.

Reply:
> For the tests test_getche and test_getch, they uses msvcrt.ungetch instead of the magic write_input, I think this stuff is stable enough

Hum, ok. Let's see if CI will agree with you :-) I'm fine with keeping them in the same process for now.